docs/advanced/document-store/redis.md

```diff
@@ -0,0 +1,339 @@
+
+- `$eq` - Equal to (number, string)
+- `$ne` - Not equal to (number, string)
+- `$gt` - Greater than (number)
+- `$gte` - Greater than or equal to (number)
+- `$lt` - Less than (number)
+- `$lte` - Less than or equal to (number)
+
+
+Consider Documents with embeddings `[0,0,0]` up to `[9,9,9]` where the document with embedding `[i,i,i]`
+has tag `price` with number value and tag `color` with string value. We can create such example with the following code:
+
+```python
+import numpy as np
+from docarray import Document, DocumentArray
+
+n_dim = 3
+
+da = DocumentArray(
+    storage='redis',
+    config={
+        'n_dim': n_dim,
+        'columns': [('price', 'int'), ('color', 'str')],
+        'flush': True,
+    },
+)
+
+da.extend(
+    [
+        Document(
+            id=f'{i}', embedding=i * np.ones(n_dim), tags={'price': i, 'color': 'red'}
+        )
+        for i in range(10)
+    ]
+)
+da.extend(
+    [
+        Document(
+            id=f'{i+10}',
+            embedding=i * np.ones(n_dim),
+            tags={'price': i, 'color': 'blue'},
+        )
+        for i in range(10)
+    ]
+)
+
+print('\nIndexed prices and colors:\n')
+for embedding, price, color in zip(
+    da.embeddings, da[:, 'tags__price'], da[:, 'tags__color']
+):
+    print(f'\tembedding={embedding},\t price={price},\t color={color}')
+```
+
+Consider we want the nearest vectors to the embedding `[8. 8. 8.]`, with the restriction that
+prices and color must follow a filter. For example, let's consider that retrieved documents must have a `price` value lower than or equal to `max_price` and have `color` equal to `color`. We can encode this information in Redis using `{'price': {'$lte': max_price}, 'color': {'$eq': color}}`.
+
+Then the search with the proposed filter can be used as follows:
+```python
+max_price = 7
+color = 'red'
+n_limit = 5
+
+np_query = np.ones(n_dim) * 8
+print(f'\nQuery vector: \t{np_query}')
+
+filter = {'price': {'$lte': max_price}, 'color': {'$eq': color}}
+results = da.find(np_query, filter=filter, limit=n_limit)
+
+print(
+    '\nEmbeddings Approximate Nearest Neighbours with "price" at most 7 and "color" red:\n'
+)
+for embedding, price, color, score in zip(
+    results.embeddings,
+    results[:, 'tags__price'],
+    results[:, 'tags__color'],
+    results[:, 'scores'],
+):
+    print(
+        f' score={score["score"].value},\t embedding={embedding},\t price={price},\t color={color}'
+    )
+```
+
+This would print:
+
+```console
+Embeddings Approximate Nearest Neighbours with "price" at most 7 and "color" red:
+
+ embedding=[3. 3. 3.],   price=3,        color=red,      score=0
+ embedding=[6. 6. 6.],   price=6,        color=red,      score=0
+ embedding=[1. 1. 1.],   price=1,        color=red,      score=5.96046447754e-08
+ embedding=[2. 2. 2.],   price=2,        color=red,      score=5.96046447754e-08
+ embedding=[4. 4. 4.],   price=4,        color=red,      score=5.96046447754e-08
```

Comment:
something is wrong, shouldn't we find those closes to (8, 8, 8) ? (or maybe we're not putting the right values as example)